engines/macventure/macventure.cpp

```diff
@@ -0,0 +1,1184 @@
+}
+
+void MacVentureEngine::endGame() {
+	requestQuit();
+}
+
+void MacVentureEngine::updateState(bool pause) {
+	_prepared = false;
+	runObjQueue();
+	printTexts();
+	playSounds(pause);
+}
+
+void MacVentureEngine::revert() {
+	_gui->invertWindowColors(kMainGameWindow);
+	preparedToRun();
+}
+
+void MacVentureEngine::runObjQueue() {
+	while (!_objQueue.empty()) {
+		uint32 biggest = 0;
+		uint32 index = 0;
+		uint32 temp;
+		for (uint i = 0; i < _objQueue.size(); i++) {
+			temp = _objQueue[i].id;
+			if (temp > biggest) {
+				biggest = temp;
+				index = i;
+			}
+		}
+		QueuedObject obj = _objQueue[index];
+		_objQueue.remove_at(index);
+		switch (obj.id) {
+		case 0x2:
+			focusObjectWindow(obj.object);
+			break;
+		case 0x3:
+			openObject(obj.object);
+			break;
+		case 0x4:
+			closeObject(obj.object);
+			break;
+		case 0x7:
+			checkObject(obj);
+			break;
+		case 0x8:
+			reflectSwap(obj.object, obj.target);
+			break;
+		case 0xc:
+			_world->setObjAttr(_gui->getWindowData(kMainGameWindow).refcon, kAttrContainerOpen, 0);
+			_world->setObjAttr(_world->getObjAttr(1, kAttrParentObject), kAttrContainerOpen, 1);
+			break;
+		case 0xd:
+			toggleExits();
+			break;
+		case 0xe:
+			zoomObject(obj.object);
+			break;
+		}
+	}
+}
+
+void MacVentureEngine::printTexts() {
+	for (uint i = 0; i < _textQueue.size(); i++) {
+		QueuedText text = _textQueue.front();
+		_textQueue.remove_at(0);
+		switch (text.id) {
+		case kTextNumber:
+			_gui->printText(Common::String(text.asset));
+			gameChanged();
+			break;
+		case kTextNewLine:
+			_gui->printText(Common::String(""));
+			gameChanged();
+			break;
+		case kTextPlain:
+			_gui->printText(_world->getText(text.asset, text.source, text.destination));
+			gameChanged();
+			break;
+		}
+	}
+}
+
+void MacVentureEngine::playSounds(bool pause) {
+	int delay=0;
+	while (!_soundQueue.empty()) {
+		QueuedSound item = _soundQueue.front();
+		_soundQueue.remove_at(0);
+		switch (item.id) {
+		case kSoundPlay:
+			_soundManager->playSound(item.reference);
+			break;
+		case kSoundPlayAndWait:
+			delay = _soundManager->playSound(item.reference);
+			break;
+		case kSoundWait:
+			//wait for sound to finish?
+			break;
+		}
+	}
+	if (pause && delay > 0) {
+		warning("Sound pausing not yet tested. Pausing for %d", delay);
+		g_system->delayMillis(delay);
+		preparedToRun();
+	}
+}
+
+void MacVentureEngine::updateControls() {
+	selectControl(kNoCommand);
+	_gui->clearControls();
+	toggleExits();
+	resetVars();
+}
+
+void MacVentureEngine::resetVars() {
+	selectControl(kNoCommand);
+	_currentSelection.clear();
+	_destObject = 0;
+	setDeltaPoint(Common::Point(0, 0));
+	_cmdReady = false;
+}
+
+void MacVentureEngine::unselectAll() {
+	while (!_currentSelection.empty()) {
+		unselectObject(_currentSelection.front());
+	}
+}
+
+void MacVentureEngine::selectObject(ObjID objID) {
+	if (!_currentSelection.empty()) {
+		if (findParentWindow(objID) != findParentWindow(_currentSelection[0])) {
+			//unselectAll();
```

Comment:
I recommend to put a TODO here, as it will be easier to search for not finished places. Same for the other similar occurences.